### PR TITLE
Upgrade blog components to Docusaurus beta15

### PR DESCRIPTION
**Pull request by ivarpruijn**
- Upgrade blog components to Docusaurus beta15. Only some internal refactoring by Docusaurus, nothing that was broken.
- Remove `theme-classic` dependency, as it's already imported by the `preset-classic` dependency.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,6 @@
   "dependencies": {
     "@docusaurus/core": "^2.0.0-beta.15",
     "@docusaurus/preset-classic": "^2.0.0-beta.15",
-    "@docusaurus/theme-classic": "^2.0.0-beta.15",
```
Comment (sdirosa) [inline package.json:20]: Nice 👌🏼

**Comment by sdirosa** [comment]
@ivarpruijn There are other dependencies that are fairly outdated. Do we want to bump those up in this PR or perhaps in a clean one?

**Comment by ivarpruijn** [comment]
> @ivarpruijn There are other dependencies that are fairly outdated. Do we want to bump those up in this PR or perhaps in a clean one?

Good one, but think we should do that in another PR then.

File: src/theme/BlogPostItem/index.tsx [hunk 2]
```diff
@@ -77,17 +78,25 @@ function BlogPostItem(props: Props): JSX.Element {
...
+      itemScope
+      itemType="http://schema.org/BlogPosting">
       <TrackedHeader>
         <TitleHeading className={styles.blogPostTitle} itemProp="headline">
           {isBlogPostPage ? (
             title
           ) : (
-            <TrackedLink itemProp="url" to={permalink}>
+            <TrackedLink 
+              id={`post-${makeIdFromString(title)}`}
```
Comment (sdirosa) [inline src/theme/BlogPostItem/index.tsx:97]: This identifier should not be unique, the ContentContext around this whole Component already describes in which post we are and the TrackedHeader ensures other links, identical to this one, will not collide with it.

File: src/theme/PaginatorNavLink/index.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+function PaginatorNavLink(props: Props): JSX.Element {
+  const {permalink, title, subLabel} = props;
+  return (
+    <TrackedLink className="pagination-nav__link" to={permalink}>
+      {subLabel && <div className="pagination-nav__sublabel">{subLabel}</div>}
+      <div className="pagination-nav__label">{title}</div>
+    </TrackedLink>
+  );
+}
```
Comment (sdirosa) [inline src/theme/PaginatorNavLink/index.tsx:21]: Nice Component!

File: src/theme/BlogPostItem/index.tsx (hunk 2 above)
Comment (ivarpruijn) [reply src/theme/BlogPostItem/index.tsx:97]: You're right. I was testing with a 2nd blog post with the exact same title, but of course that collides. That will never happen in practice, so I'll remove this ID.

Comment (sdirosa) [reply src/theme/BlogPostItem/index.tsx]: Indeed, that's the only scenario but it's really a strange case imho, the routing wouldn't work either and the second article would never be reachable

**Review by sdirosa — APPROVED** [review]

**Comment by ivarpruijn** [comment]
Tested on staging, ran broken links checker, all lgtm

**Pull request merged**